taskvine/src/manager/vine_manager.c

```diff
@@ -3269,13 +3280,15 @@ static int send_one_task(struct vine_manager *q)
 				continue;
 		}
 
-		// Otherwise, remove it from the ready list and start it:
-		list_pop_tail(q->ready_list);
+		// Otherwise, remove it from the ready queue and start it:
+		priority_queue_remove(q->ready_tasks, t_idx);
+
 		commit_task_to_worker(q, w, t);
 		return 1;
 	}
+	PRIORITY_QUEUE_ROTATE_ITERATE(q->ready_tasks, t_idx, t);
```

Comment:
Yikes, this code is really confusing because the macro evaluates to a `while` that modifies the loop above it.  I think every person who looks at this code is going to be confused.  Can you find a way to make its purpose more obvious?  For example, can you express the same code like this:

```
SOME_MACRO(queue, index,task) {
   /* body of loop */
}
```